lib/prelude.dx

```diff
@@ -1221,6 +1253,11 @@ instance Arbitrary Int32
 instance [Arbitrary a] Arbitrary (n=>a)
   arb = \key. for i. arb $ ixkey key i
 
+instance [Arbitrary a, Arbitrary b] Arbitrary (a & b)
```

Comment:
Glad to see you're finding this type class useful!

Reply:
These are great. I like the new splitKey as well. 

I had written something similar for neural network initialization. Although for NN's there are some strange hacks like scaling based on matrix size, so they might need their own initializers. (Dex doesn't have newtypes right?)

Reply:
Dex doesn't have newtypes, but I think you get the same functionality with ordinary `data` declarations. The main benefit of Haskell's `newtype` is that it's guaranteed to be zero-cost at runtime but that's true of Dex's `data` wrappers anyway. Haskell's `newtype` also allows things like `coerce` and fancy versions of `deriving` but Dex doesn't have those yet.